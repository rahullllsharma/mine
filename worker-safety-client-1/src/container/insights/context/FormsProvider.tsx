import type { ReactNode } from "react";
import type { FormFilter } from "../filters/formFilters/FormFilters";
import { noop } from "lodash-es";
import {
  createContext,
  useCallback,
  useContext,
  useMemo,
  useState,
} from "react";

type FormProviderProps = {
  children: ReactNode;
  defaultFilters?: FormFilter[];
};

type FormContextProps = {
  filters: FormFilter[];
  setFilters: (filters: FormFilter[]) => void;
};

//defualt filters
const defaultState: FormContextProps = {
  filters: [
    { field: "STATUS", values: [] },
    { field: "FORM", values: [] },
    { field: "WORKPACKAGE", values: [] },
    { field: "CREATEDBY", values: [] },
    { field: "CREATEDON", values: [] },
    { field: "UPDATEDBY", values: [] },
    { field: "UPDATEDON", values: [] },
    { field: "LOCATIONS", values: [] },
    { field: "COMPLETEDON", values: [] },
    { field: "REPORTDATE", values: [] },
  ],
  setFilters: noop,
};

const FormContext = createContext<FormContextProps>(defaultState);

export const useFormContext = (): FormContextProps => useContext(FormContext);

export default function FormProvider({
  children,
  defaultFilters = defaultState.filters,
}: FormProviderProps): JSX.Element {
  const [formFilters, setFormFilters] = useState(defaultFilters);
  const setFilters = useCallback((filters: FormFilter[]) => {
    setFormFilters(filters);
  }, []);

  const formContext: FormContextProps = useMemo(
    () => ({
      filters: formFilters,
      setFilters,
    }),
    [formFilters, setFilters]
  );

  return (
    <FormContext.Provider value={formContext}>{children}</FormContext.Provider>
  );
}
