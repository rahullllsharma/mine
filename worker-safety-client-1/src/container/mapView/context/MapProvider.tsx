import type { ReactNode } from "react";
import type { LocationFilter } from "../filters/mapFilters/MapFilters";
import { noop } from "lodash-es";
import {
  createContext,
  useCallback,
  useContext,
  useEffect,
  useMemo,
  useState,
} from "react";
import { useAuthStore } from "@/store/auth/useAuthStore.store";

type MapProviderProps = {
  children: ReactNode;
  defaultFilters?: LocationFilter[];
};

type MapContextItemProps = {
  isActive: boolean;
  toggle: () => void;
};

type MapContextProps = {
  riskLegend: MapContextItemProps;
  satelliteView: MapContextItemProps;
  filters: LocationFilter[];
  setFilters: (filters: LocationFilter[]) => void;
};

const defaultFilters: LocationFilter[] = [
  { field: "RISK", values: [] },
  { field: "TYPES", values: [] },
  { field: "DIVISIONS", values: [] },
  { field: "REGIONS", values: [] },
  { field: "PROJECT", values: [] },
  { field: "CONTRACTOR", values: [] },
  { field: "SUPERVISOR", values: [] },
  { field: "PROJECT_STATUS", values: [] },
];

const defaultState: MapContextProps = {
  riskLegend: { isActive: true, toggle: noop },
  satelliteView: { isActive: false, toggle: noop },
  filters: defaultFilters,
  setFilters: noop,
};

const MapContext = createContext<MapContextProps>(defaultState);

export const useMapContext = (): MapContextProps => useContext(MapContext);

export default function MapProvider({
  children,
  defaultFilters: fallbackFilters = defaultFilters,
}: MapProviderProps): JSX.Element {
  const [isRiskLevelVisible, setIsRiskLevelVisible] = useState(true);
  const [isSatelliteViewEnabled, setIsSatelliteViewEnabled] = useState(false);

  const [mapFilters, setMapFilters] = useState<LocationFilter[]>([]);
  const { getLocationMapFilters, setLocationMapFilters } = useAuthStore();

  useEffect(() => {
    const stored = getLocationMapFilters();
    setMapFilters(stored?.length ? stored : fallbackFilters);
  }, [fallbackFilters, getLocationMapFilters]);

  const toggleRiskLevel = useCallback(() => {
    setIsRiskLevelVisible(prevState => !prevState);
  }, []);

  const toggleSatelliteView = useCallback(() => {
    setIsSatelliteViewEnabled(prevState => !prevState);
  }, []);

  const setFilters = useCallback(
    (filters: LocationFilter[]) => {
      setMapFilters(filters);
      setLocationMapFilters(filters);
    },
    [setLocationMapFilters]
  );

  const mapContext = useMemo<MapContextProps>(
    () => ({
      riskLegend: { isActive: isRiskLevelVisible, toggle: toggleRiskLevel },
      satelliteView: {
        isActive: isSatelliteViewEnabled,
        toggle: toggleSatelliteView,
      },
      filters: mapFilters,
      setFilters,
    }),
    [
      isRiskLevelVisible,
      toggleRiskLevel,
      isSatelliteViewEnabled,
      toggleSatelliteView,
      mapFilters,
      setFilters,
    ]
  );

  return (
    <MapContext.Provider value={mapContext}>{children}</MapContext.Provider>
  );
}
