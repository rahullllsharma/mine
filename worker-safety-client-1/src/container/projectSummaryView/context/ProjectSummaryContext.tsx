import type { ReactNode } from "react";
import { noop } from "lodash-es";
import { useContext, createContext } from "react";

type ProjectSummaryContext = {
  refetchActivitiesBasedOnLocation: () => void;
};

type ProjectSummaryViewProviderProps = ProjectSummaryContext & {
  children: ReactNode;
};

const ProjectSummaryViewContext = createContext<ProjectSummaryContext>({
  refetchActivitiesBasedOnLocation: noop,
});

const useProjectSummaryEvents = () => useContext(ProjectSummaryViewContext);

const ProjectSummaryViewProvider = ({
  children,
  refetchActivitiesBasedOnLocation,
}: ProjectSummaryViewProviderProps) => {
  const context = {
    refetchActivitiesBasedOnLocation,
  };

  return (
    <ProjectSummaryViewContext.Provider value={context}>
      {children}
    </ProjectSummaryViewContext.Provider>
  );
};

export type { ProjectSummaryContext };
export {
  ProjectSummaryViewContext,
  ProjectSummaryViewProvider,
  useProjectSummaryEvents,
};
