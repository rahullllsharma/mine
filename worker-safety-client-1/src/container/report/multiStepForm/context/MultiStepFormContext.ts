import type {
  MultiStepFormActions,
  MultiStepFormState,
} from "../state/reducer";

import type { Dispatch } from "react";
import { createContext, useContext } from "react";

const MultiStepFormContext = createContext<
  [MultiStepFormState, Dispatch<MultiStepFormActions>] | []
>([]);

export const useMultiStepFormContext = (): [
  MultiStepFormState,
  Dispatch<MultiStepFormActions>
] =>
  useContext(MultiStepFormContext) as unknown as [
    MultiStepFormState,
    Dispatch<MultiStepFormActions>
  ];

export default MultiStepFormContext;
