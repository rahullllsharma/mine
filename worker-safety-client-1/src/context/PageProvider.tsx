import type { ReactNode } from "react";
import { useContext, createContext } from "react";

type GenericProp = Record<string, unknown>;

type PageProviderProps<T extends GenericProp> = {
  children: ReactNode;
  props: T;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const PageContext = createContext<any>({});

function usePageContext<T>() {
  return useContext<T>(PageContext);
}

function PageProvider<T extends GenericProp>({
  props,
  children,
}: PageProviderProps<T>) {
  return (
    <PageContext.Provider value={{ ...props }}>{children}</PageContext.Provider>
  );
}

export { PageProvider, usePageContext };
