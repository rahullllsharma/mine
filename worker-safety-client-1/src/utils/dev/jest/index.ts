export * from "./intl.utils";
export * from "./testing.utils";
export * from "./charts.utils";
