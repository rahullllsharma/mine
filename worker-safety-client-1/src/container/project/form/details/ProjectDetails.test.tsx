import { formRender, mockTenantStore } from "@/utils/dev/jest";
import { PageProvider } from "@/context/PageProvider";
import ProjectDetails from "./ProjectDetails";

describe(ProjectDetails.name, () => {
  mockTenantStore();

  it("should render correctly", () => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    const { asFragment } = formRender(
      <PageProvider
        props={{
          divisionsLibrary: [],
          regionsLibrary: [],
          projectTypesLibrary: [],
          assetTypesLibrary: [],
          managers: [],
          supervisors: [],
          contractors: [],
        }}
      >
        <ProjectDetails />
      </PageProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
