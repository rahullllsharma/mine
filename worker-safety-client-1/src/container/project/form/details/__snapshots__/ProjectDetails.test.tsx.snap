// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectDetails should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  pointer-events: none;
  position: relative;
  box-sizing: border-box;
}

.emotion-1 {
  z-index: 9999;
  border: 0;
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  width: 1px;
  position: absolute;
  overflow: hidden;
  padding: 0;
  white-space: nowrap;
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: grid;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 2px 0 0 0;
  -webkit-overflow-scrolling: touch;
  position: relative;
  overflow: hidden;
  box-sizing: border-box;
  gap: 4px;
}

.emotion-5 {
  color: hsl(0, 0%, 50%);
  grid-area: 1/1/2/3;
  margin-left: 2px;
  margin-right: 2px;
  box-sizing: border-box;
  margin: 0;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.emotion-6 {
  margin: 0;
  padding-bottom: 2px;
  padding-top: 2px;
  visibility: hidden;
  color: hsl(0, 0%, 20%);
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: inline-grid;
  grid-area: 1/1/2/3;
  grid-template-columns: 0 min-content;
  box-sizing: border-box;
}

.emotion-6:after {
  content: attr(data-value) " ";
  visibility: hidden;
  white-space: pre;
  grid-area: 1/2;
  font: inherit;
  min-width: 2px;
  border: 0;
  margin: 0;
  outline: 0;
  padding: 0;
}

.emotion-7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  box-sizing: border-box;
}

.emotion-8 {
  color: hsl(0, 0%, 80%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0px 8px;
  -webkit-transition: color 150ms;
  transition: color 150ms;
  box-sizing: border-box;
}

.emotion-8:hover {
  color: hsl(0, 0%, 60%);
}

.emotion-24 {
  background: 0;
  border: 0;
  caret-color: transparent;
  font-size: inherit;
  grid-area: 1/1/2/3;
  outline: 0;
  padding: 0;
  width: 1px;
  color: transparent;
  left: -100px;
  opacity: 0;
  position: relative;
  -webkit-transform: scale(.01);
  -moz-transform: scale(.01);
  -ms-transform: scale(.01);
  transform: scale(.01);
}

<div>
    <h6
      data-testid="work-package-details"
    >
      Project Details
    </h6>
    <div
      class="mt-4"
    >
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-name"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="name"
          >
            Project Name *
          </label>
          <div
            class="relative flex w-full items-center text-base font-normal text-neutral-shade-100 rounded-md border border-solid border-neutral-shade-26 rounded-md focus-within:ring-1 focus-within:ring-brand-gray-60 bg-neutral-light-100"
          >
            <input
              aria-errormessage=""
              aria-invalid="false"
              class="flex-auto rounded-md appearance-none focus:outline-none disabled:bg-neutral-light-77 read-only:cursor-default disabled:cursor-not-allowed disabled:opacity-38 min-w-0 p-2"
              id="name"
              name="name"
              placeholder="Project Name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-external-key"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="externalKey"
          >
            Project Number *
          </label>
          <div
            class="relative flex w-full items-center text-base font-normal text-neutral-shade-100 rounded-md border border-solid border-neutral-shade-26 rounded-md focus-within:ring-1 focus-within:ring-brand-gray-60 bg-neutral-light-100"
          >
            <input
              aria-errormessage=""
              aria-invalid="false"
              class="flex-auto rounded-md appearance-none focus:outline-none disabled:bg-neutral-light-77 read-only:cursor-default disabled:cursor-not-allowed disabled:opacity-38 min-w-0 p-2"
              id="externalKey"
              name="externalKey"
              placeholder="Ex. 3591283"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-type"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="libraryProjectTypeId"
          >
            Project Type *
          </label>
          <div
            class="relative text-neutral-shade-100"
            data-testid="select"
          >
            <button
              aria-haspopup="true"
              class="w-full px-2 py-0.5 flex items-center justify-between bg-white text-base rounded border border-solid outline-none focus-within:ring-1 focus-within:ring-brand-gray-60 border-neutral-shade-26 text-neutral-shade-58 bg-neutral-shade-7 cursor-not-allowed"
              disabled=""
              id="headlessui-listbox-button-1"
              type="button"
            >
              <span
                class="truncate"
              >
                No options available
              </span>
              <i
                aria-hidden="true"
                class="ci-chevron_down text-neutral-shade-58 text-xl"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-asset-type"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="libraryAssetTypeId"
          >
            Asset Type
          </label>
          <div
            class="relative text-neutral-shade-100"
            data-testid="select"
          >
            <button
              aria-haspopup="true"
              class="w-full px-2 py-0.5 flex items-center justify-between bg-white text-base rounded border border-solid outline-none focus-within:ring-1 focus-within:ring-brand-gray-60 border-neutral-shade-26 text-neutral-shade-58 bg-neutral-shade-7 cursor-not-allowed"
              disabled=""
              id="headlessui-listbox-button-3"
              type="button"
            >
              <span
                class="truncate"
              >
                No options available
              </span>
              <i
                aria-hidden="true"
                class="ci-chevron_down text-neutral-shade-58 text-xl"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-status"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="status"
          >
            Project Status *
          </label>
          <div
            class="relative text-neutral-shade-100"
            data-testid="select"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              class="w-full px-2 py-0.5 flex items-center justify-between bg-white text-base rounded border border-solid outline-none focus-within:ring-1 focus-within:ring-brand-gray-60 border-neutral-shade-26 text-neutral-shade-58"
              id="headlessui-listbox-button-5"
              type="button"
            >
              <span
                class="truncate"
              >
                Select an option
              </span>
              <i
                aria-hidden="true"
                class="ci-chevron_down text-neutral-shade-58 text-xl"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-zip-code"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="projectZipCode"
          >
            Project Zip Code
          </label>
          <div
            class="relative flex w-full items-center text-base font-normal text-neutral-shade-100 rounded-md border border-solid border-neutral-shade-26 rounded-md focus-within:ring-1 focus-within:ring-brand-gray-60 bg-neutral-light-100"
          >
            <input
              aria-errormessage=""
              aria-invalid="false"
              class="flex-auto rounded-md appearance-none focus:outline-none disabled:bg-neutral-light-77 read-only:cursor-default disabled:cursor-not-allowed disabled:opacity-38 min-w-0 p-2"
              id="projectZipCode"
              name="projectZipCode"
              placeholder="Ex. 10010"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="flex pb-4 gap-4"
      >
        <div
          class="flex flex-1 flex-col"
          data-testid="work-package-start-date"
        >
          <div>
            <label
              class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
              for="startDate"
            >
              Start Date *
            </label>
            <div
              class="relative flex w-full items-center text-base font-normal text-neutral-shade-100 rounded-md border border-solid text-neutral-shade-58 border-neutral-shade-26 rounded-md focus-within:ring-1 focus-within:ring-brand-gray-60 bg-neutral-light-100"
            >
              <i
                aria-hidden="true"
                class="ci-calendar ml-2 pointer-events-none w-6 h-6 text-xl leading-none bg-white text-neutral-shade-58"
              />
              <input
                aria-errormessage=""
                aria-invalid="false"
                class="flex-auto rounded-md appearance-none focus:outline-none disabled:bg-neutral-light-77 read-only:cursor-default disabled:cursor-not-allowed disabled:opacity-38 min-w-0 root p-2"
                id="startDate"
                name="startDate"
                type="date"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="flex flex-1 flex-col"
          data-testid="work-package-end-date"
        >
          <div>
            <label
              class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
              for="endDate"
            >
              End Date *
            </label>
            <div
              class="relative flex w-full items-center text-base font-normal text-neutral-shade-100 rounded-md border border-solid text-neutral-shade-58 border-neutral-shade-26 rounded-md focus-within:ring-1 focus-within:ring-brand-gray-60 bg-neutral-light-100"
            >
              <i
                aria-hidden="true"
                class="ci-calendar ml-2 pointer-events-none w-6 h-6 text-xl leading-none bg-white text-neutral-shade-58"
              />
              <input
                aria-errormessage=""
                aria-invalid="false"
                class="flex-auto rounded-md appearance-none focus:outline-none disabled:bg-neutral-light-77 read-only:cursor-default disabled:cursor-not-allowed disabled:opacity-38 min-w-0 root p-2"
                id="endDate"
                name="endDate"
                type="date"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-division"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="libraryDivisionId"
          >
            Division *
          </label>
          <div
            class="relative text-neutral-shade-100"
            data-testid="select"
          >
            <button
              aria-haspopup="true"
              class="w-full px-2 py-0.5 flex items-center justify-between bg-white text-base rounded border border-solid outline-none focus-within:ring-1 focus-within:ring-brand-gray-60 border-neutral-shade-26 text-neutral-shade-58 bg-neutral-shade-7 cursor-not-allowed"
              disabled=""
              id="headlessui-listbox-button-7"
              type="button"
            >
              <span
                class="truncate"
              >
                No options available
              </span>
              <i
                aria-hidden="true"
                class="ci-chevron_down text-neutral-shade-58 text-xl"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-region"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="libraryRegionId"
          >
            Region *
          </label>
          <div
            class="relative text-neutral-shade-100"
            data-testid="select"
          >
            <button
              aria-haspopup="true"
              class="w-full px-2 py-0.5 flex items-center justify-between bg-white text-base rounded border border-solid outline-none focus-within:ring-1 focus-within:ring-brand-gray-60 border-neutral-shade-26 text-neutral-shade-58 bg-neutral-shade-7 cursor-not-allowed"
              disabled=""
              id="headlessui-listbox-button-9"
              type="button"
            >
              <span
                class="truncate"
              >
                No options available
              </span>
              <i
                aria-hidden="true"
                class="ci-chevron_down text-neutral-shade-58 text-xl"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-description"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="description"
          >
            Description
          </label>
          <textarea
            aria-label="FieldTextArea"
            class="w-full max-h-36 text-base text-neutral-shade-100 outline-none focus:border-brand-gray-60 disabled:cursor-not-allowed min-h-12 border border-neutral-shade-37 rounded resize-y py-1 px-2"
            id="description"
            name="description"
            placeholder="Add a description..."
          />
        </div>
      </div>
      <h6
        class="py-4"
        data-testid="work-package-team-members"
      >
        Project Team Members
      </h6>
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-manager"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="projectManager"
          >
            Project Manager *
          </label>
          <div
            class=" emotion-0"
          >
            <span
              class="emotion-1"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="emotion-1"
            />
            <div
              class="w-full pr-2 py-0.5 flex items-center justify-between bg-white text-base rounded border border-solid outline-none focus-within:ring-1 focus-within:ring-brand-gray-60 cursor-pointer border-neutral-shade-26 text-neutral-shade-58 bg-neutral-shade-7 cursor-not-allowed emotion-3"
              role="button"
            >
              <div
                class="ml-2 mr-1 pointer-events-none w-6 h-6 text-xl leading-none bg-white opacity-38"
              >
                <i
                  aria-hidden="true"
                  class="ci-user"
                />
              </div>
              <div
                class=" emotion-4"
              >
                <div
                  class=" emotion-5"
                  id="react-select-2-placeholder"
                >
                  No options available
                </div>
                <div
                  class=" emotion-6"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-controls="react-select-2-listbox"
                    aria-describedby="react-select-2-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-owns="react-select-2-listbox"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    disabled=""
                    id="react-select-2-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" emotion-7"
              >
                <div
                  aria-hidden="true"
                  class=" emotion-8"
                >
                  <i
                    aria-hidden="true"
                    class="ci-chevron_down text-neutral-shade-58 text-xl"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-assigned-person"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="supervisor"
          >
            Primary Project Supervisor *
          </label>
          <div
            class=" emotion-0"
          >
            <span
              class="emotion-1"
              id="react-select-3-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="emotion-1"
            />
            <div
              class="w-full pr-2 py-0.5 flex items-center justify-between bg-white text-base rounded border border-solid outline-none focus-within:ring-1 focus-within:ring-brand-gray-60 cursor-pointer border-neutral-shade-26 text-neutral-shade-58 bg-neutral-shade-7 cursor-not-allowed emotion-3"
              role="button"
            >
              <div
                class="ml-2 mr-1 pointer-events-none w-6 h-6 text-xl leading-none bg-white opacity-38"
              >
                <i
                  aria-hidden="true"
                  class="ci-user"
                />
              </div>
              <div
                class=" emotion-4"
              >
                <div
                  class=" emotion-5"
                  id="react-select-3-placeholder"
                >
                  No options available
                </div>
                <div
                  class=" emotion-6"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-controls="react-select-3-listbox"
                    aria-describedby="react-select-3-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-owns="react-select-3-listbox"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    disabled=""
                    id="react-select-3-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" emotion-7"
              >
                <div
                  aria-hidden="true"
                  class=" emotion-8"
                >
                  <i
                    aria-hidden="true"
                    class="ci-chevron_down text-neutral-shade-58 text-xl"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-additional-assigned-person"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="additionalSupervisors"
          >
            Additional Project Supervisor(s)
          </label>
          <div
            class=" emotion-0"
          >
            <span
              class="emotion-1"
              id="react-select-4-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="emotion-1"
            />
            <div
              class="w-full pr-2 py-0.5 flex items-center justify-between bg-white text-base rounded border border-solid outline-none focus-within:ring-1 focus-within:ring-brand-gray-60 cursor-pointer border-neutral-shade-26 text-neutral-shade-58 bg-neutral-shade-7 cursor-not-allowed emotion-3"
              role="button"
            >
              <div
                class="ml-2 mr-1 pointer-events-none w-6 h-6 text-xl leading-none bg-white opacity-38"
              >
                <i
                  aria-hidden="true"
                  class="ci-user"
                />
              </div>
              <div
                class=" emotion-4"
              >
                <div
                  class=" emotion-5"
                  id="react-select-4-placeholder"
                >
                  No options available
                </div>
                <input
                  aria-autocomplete="list"
                  aria-controls="react-select-4-listbox"
                  aria-describedby="react-select-4-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-owns="react-select-4-listbox"
                  aria-readonly="true"
                  class="emotion-24"
                  disabled=""
                  id="react-select-4-input"
                  inputmode="none"
                  role="combobox"
                  tabindex="0"
                  value=""
                />
              </div>
              <div
                class=" emotion-7"
              >
                <div
                  aria-hidden="true"
                  class=" emotion-8"
                >
                  <i
                    aria-hidden="true"
                    class="ci-chevron_down text-neutral-shade-58 text-xl"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-engineer"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="engineerName"
          >
            Engineer
          </label>
          <div
            class="relative flex w-full items-center text-base font-normal text-neutral-shade-100 rounded-md border border-solid border-neutral-shade-26 rounded-md focus-within:ring-1 focus-within:ring-brand-gray-60 bg-neutral-light-100"
          >
            <input
              aria-errormessage=""
              aria-invalid="false"
              class="flex-auto rounded-md appearance-none focus:outline-none disabled:bg-neutral-light-77 read-only:cursor-default disabled:cursor-not-allowed disabled:opacity-38 min-w-0 p-2"
              id="engineerName"
              name="engineerName"
              placeholder="Engineer name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <h6
        class="py-4"
        data-testid="work-package-contract-information"
      >
        Contract Information
      </h6>
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-contract-reference-number"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="contractReference"
          >
            Contract Reference Number
          </label>
          <div
            class="relative flex w-full items-center text-base font-normal text-neutral-shade-100 rounded-md border border-solid border-neutral-shade-26 rounded-md focus-within:ring-1 focus-within:ring-brand-gray-60 bg-neutral-light-100"
          >
            <input
              aria-errormessage=""
              aria-invalid="false"
              class="flex-auto rounded-md appearance-none focus:outline-none disabled:bg-neutral-light-77 read-only:cursor-default disabled:cursor-not-allowed disabled:opacity-38 min-w-0 p-2"
              id="contractReference"
              name="contractReference"
              placeholder="Ex. ABC-1234"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-contract-name"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="contractName"
          >
            Contract Name
          </label>
          <div
            class="relative flex w-full items-center text-base font-normal text-neutral-shade-100 rounded-md border border-solid border-neutral-shade-26 rounded-md focus-within:ring-1 focus-within:ring-brand-gray-60 bg-neutral-light-100"
          >
            <input
              aria-errormessage=""
              aria-invalid="false"
              class="flex-auto rounded-md appearance-none focus:outline-none disabled:bg-neutral-light-77 read-only:cursor-default disabled:cursor-not-allowed disabled:opacity-38 min-w-0 p-2"
              id="contractName"
              name="contractName"
              placeholder="Contract Name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="flex flex-col pb-4"
        data-testid="work-package-contractor"
      >
        <div>
          <label
            class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
            for="contractor"
          >
            Contractor *
          </label>
          <div
            class=" emotion-0"
          >
            <span
              class="emotion-1"
              id="react-select-5-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="emotion-1"
            />
            <div
              class="w-full pr-2 py-0.5 flex items-center justify-between bg-white text-base rounded border border-solid outline-none focus-within:ring-1 focus-within:ring-brand-gray-60 cursor-pointer border-neutral-shade-26 text-neutral-shade-58 bg-neutral-shade-7 cursor-not-allowed emotion-3"
              role="button"
            >
              <div
                class="ml-2 mr-1 pointer-events-none w-6 h-6 text-xl leading-none bg-white opacity-38"
              >
                <i
                  aria-hidden="true"
                  class="ci-user"
                />
              </div>
              <div
                class=" emotion-4"
              >
                <div
                  class=" emotion-5"
                  id="react-select-5-placeholder"
                >
                  No options available
                </div>
                <div
                  class=" emotion-6"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-controls="react-select-5-listbox"
                    aria-describedby="react-select-5-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-owns="react-select-5-listbox"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    disabled=""
                    id="react-select-5-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" emotion-7"
              >
                <div
                  aria-hidden="true"
                  class=" emotion-8"
                >
                  <i
                    aria-hidden="true"
                    class="ci-chevron_down text-neutral-shade-58 text-xl"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
